### PR TITLE
Adding control term

### PR DESCRIPTION
**Pull request by ayrton04**
This PR adds the ability to take in a control term to the EKF and UKF. It uses a Twist or TwistStamped as input. Users can configure which terms are used from the control message. Each velocity in the control term is converted to an acceleration and applied to the filter, thus aiding in minimizing lag in convergence, especially when fusing angular velocities.

### CODE REVIEW

**Comment by ayrton04** [comment]
FYI @bradleypowers @kayak0806 @davelkan @tappan-at-git

File: include/robot_localization/filter_base.h [hunk 1]
```diff
@@ -242,16 +261,83 @@ class FilterBase
...
+    {
+      FB_DEBUG("---------- FilterBase::computeControlAcceleration ----------\n");
+
+      const double error = control - state;
+      const bool sameSign = (::fabs(error) <= ::fabs(control) + 0.01);
+      const double setPoint = (sameSign ? control : 0.0);
+      const bool decelerating = ::fabs(setPoint) < ::fabs(state);
+      const double accelLimit = (decelerating ? decelerationLimit : accelerationLimit);
+
+      const double finalAccel = std::min(std::max(frequency * error, -accelLimit), accelLimit);
```
Comment (tappan-at-git) [inline include/robot_localization/filter_base.h]: What did you decide, re: our asymmetric positive and negative acceleration limits?

Comment (ayrton04) [inline include/robot_localization/filter_base.h]: So for actively driving the robot (i.e., we are telling the motors to move, whether forward or backward), we use acceleration limits, and just negate them for the direction in question. When we let off the controls, we use deceleration limits. For the weird case where we are moving in one direction and instantly switch to the other, we use deceleration limits until we reach zero, then acceleration limits.

In other words, motion towards zero means we use deceleration limits. Motion away from zero means we use acceleration limits. Signs are negated as necessary.

File: include/robot_localization/ros_filter.h [hunk 2]
```diff
@@ -78,7 +80,7 @@ struct CallbackData
                const double rejectionThreshold) :
     topicName_(topicName),
     updateVector_(updateVector),
-    updateSum_(updateSum_),
+    updateSum_(updateSum),
```
Comment (bradleypowers) [inline include/robot_localization/ros_filter.h]: (Not important) These fn/variable names cause me physical pain.

File: include/robot_localization/filter_base.h (hunk 1 above)
Comment (tappan-at-git) [inline include/robot_localization/filter_base.h]: Oh, blah. I see what you're doing now. I should have started on my coffee earlier. Looks good.

**Comment by tappan-at-git** [comment]
I don't see any issues, besides literally all of your variables being camel case instead of snake case. :P Not a critical part of this PR, obviously.

**Comment by bradleypowers** [comment]
:+1:

**Comment by bradleypowers** [comment]
Per  [the ROSCpp Style Guide](http://wiki.ros.org/CppStyleGuide#Function_.2BAC8_Methods) functions and methods *should* be camelCased.  This **notably does differ** from the [ROS Python style guide](http://wiki.ros.org/PyStyleGuide#Coding_Style), which is just PEP8 with 2 space tabs isntead of 4.

It is kinda unfortunate that these two differ, but differ they do.  This code is very style-compliant.

**Comment by ayrton04** [comment]
Well, my variables are camel cased too, and that *is* wrong. The problem is that fixing it would mean conflicts with every single fork out there.

**Comment by bradleypowers** [comment]
Fair.
